test/common/network/utility_test.cc

```diff
@@ -831,6 +831,36 @@ TEST_F(ExecInNetnsTest, OpenFail) {
   // Expecting failure.
   auto result = Utility::execInNetworkNamespace([]() -> int { return 0; }, "bleh");
   EXPECT_FALSE(result.ok());
+  EXPECT_TRUE(result.status().message().starts_with("failed to open netns file"));
+}
+
+TEST_F(ExecInNetnsTest, FailtoReturnToOriginalNetns) {
+  EXPECT_DEATH(
```

Comment:
Somewhat orthogonal to this PR: I'm not very familiar with the internal design, but I suggest considering alternatives to death in some of these cases. I assume this is not happening after the config ingestion, and maybe one can mark a listener as broken/unhealthy instead of killing the entire proxy.